### PR TITLE
Add generated policy to remove needs-triage when an area is added

### LINKED ISSUE #3340: Update bot to remove needs triage label once an area is assigned
### Clear and concise description of the problem

We should update our bot to remove the `needs triage` label when an area is assigned.

We should also re-add the `needs triage` if all area labels are removed.

### Checklist

- [X] Follow our [Code of Conduct](https://github.com/microsoft/typespec/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [docs](https://typespec.io/docs/).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### PR DESCRIPTION
**Pull request by timotheeguerin**
Generate the Microsoft policy service files to automatically to:
- add/remove `needs-triage` automatically when there is no area or an area is added
- add the area on PRs

resolve #3340

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -224,12 +224,31 @@ As a contributor you can run the following command to trigger the build and crea
 
 # Issue and Pr processes
 
+## Triage process
+
+Each team might use their own way of triaging issues however figuring out the area is a common process.
+
+```mermaid
```
Comment (timotheeguerin) [inline CONTRIBUTING.md:231]: Update with latest graph if everyone is onboard

File: .github/policies/issues.triage.yml [hunk 2]
```diff
@@ -1,26 +1,139 @@
...
-onFailure:
-onSuccess:
+      - description: Remove `needs-triage` label when an area label is added
+        if:
+          - payloadType: Issues
+          - hasLabel:
+              label: needs-triage
+          - isOpen
+          - or:
+              - labelAdded:
```
Comment (m-nash) [inline .github/policies/issues.triage.yml:28]: any chance we can define this in a common file so its not duplicated?

File: .github/policies/pr.triage.yml [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+disabled: false
+configuration:
+  resourceManagementConfiguration:
+    eventResponderTasks:
+      - description: Assign area labels to PRs based on modified files
+        if:
+          - payloadType: Pull_Request
+          - isOpen
+        then:
+          - if:
```
Comment (m-nash) [inline .github/policies/pr.triage.yml:15]: We have this mapping in code as well of area to label, any chance we can change this yml to call some code that uses the same definition so we don't have it in two places?

File: .github/policies/issues.triage.yml (hunk 2 above)
Comment (m-nash) [reply .github/policies/issues.triage.yml:28]: Spoke offline these are generated files so they are using the common definition.  I suggest we rename to indicate it is generated to make it easier to know.

File: .github/policies/pr.triage.yml (hunk 3 above)
Comment (m-nash) [reply .github/policies/pr.triage.yml:15]: same as above

**Review by m-nash — APPROVED** [review]

**Review by markcowl — APPROVED** [review]

File: .github/policies/issues.triage.generated.yml [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+              - labelAdded:
+                  label: lib:versioning
+              - labelAdded:
+                  label: meta:blog
+              - labelAdded:
+                  label: meta:website
+              - labelAdded:
+                  label: tspd
+              - labelAdded:
+                  label: emitter:client:csharp
```
[the remark this answers is not in the document]
Comment (timotheeguerin) [reply .github/policies/issues.triage.generated.yml:49]: talked offline, I'll make a follow up soon with any fix if this doesn't work and add it.

**Pull request merged**